omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,303 @@
+
+            savedIdentifierSource =  save(identifierSource);	
+            }
+        else if(identifierSourceType.equals("RemoteIdentifierSource")) {
+
+            RemoteIdentifierSource identifierSource = new RemoteIdentifierSource();
+            String username = validateInput(postBody.get("username"));
+            String password = validateInput(postBody.get("password"));
+            String url = validateInput(postBody.get("url"));
+
+            if (!StringUtils.isBlank(description)) {
+                identifierSource.setDescription(description);
+            }
+            if (StringUtils.isBlank(url)) { 
+                    errors.add("url"); 
+            }
+            if (!StringUtils.isBlank(username)) { 
+                    identifierSource.setUser(username); 
+            }
+            if (!StringUtils.isBlank(password)) {
+                if(StringUtils.isBlank(username)) {
+                    errors.add("username");
+                }
+                else {
+                    identifierSource.setPassword(password);
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ConversionException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name);
+            identifierSource.setUrl(url);
+            identifierSource.setDescription(description);
+
+            savedIdentifierSource =  save(identifierSource);
+
+        }
+        else if(identifierSourceType.equals("IdentifierPool")){
+
+            IdentifierPool identifierSource = new IdentifierPool();
+            String batchSize = validateInput(postBody.get("batchSize"));
+            String minPoolSize = validateInput(postBody.get("minPoolSize"));
+            String sourceUuid = validateInput(postBody.get("sourceUuid"));
+            String sequential = validateInput(postBody.get("sequential"));
+            String refillWithScheduledTask = validateInput(postBody.get("refillWithScheduledTask"));
+            IdentifierSource poolSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid);
+
+            if (!StringUtils.isBlank(description)) {
+                identifierSource.setDescription(description);
+            }
+            if (!StringUtils.isBlank(batchSize)) {
+                if(StringUtils.isNumeric(batchSize)) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize));
+                }else {
+                    errors.add("batch size");
+                }
+            }
+            if (!StringUtils.isBlank(minPoolSize)) {
+                if(StringUtils.isNumeric(minPoolSize)) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize));
+                    Set<PooledIdentifier> pooledIdentifiers = new LinkedHashSet<PooledIdentifier>();
```

Comment:
Do we really need to do this setting of identifiers?
Did you see it done anywhere in the existing IdentifierSourceController that used to handle the old UI?

Reply:
Fixed. It is much simpler than I thought: [Add Pool](https://github.com/openmrs/openmrs-module-idgen/blob/master/omod/src/main/java/org/openmrs/module/idgen/web/controller/IdentifierSourceController.java#L269). Thanks.

Reply:
I do not think you need to call that method.
You only need to validate as here https://github.com/openmrs/openmrs-module-idgen/blob/master/omod/src/main/java/org/openmrs/module/idgen/web/controller/IdentifierSourceController.java#L152-L166 and then save.

Reply:
fixed